src/mono/mono/mini/mini-amd64.c

```diff
@@ -1008,6 +1009,34 @@ get_call_info (MonoMemPool *mp, MonoMethodSignature *sig)
 		}
 
 		ptype = mini_get_underlying_type (sig->params [i]);
+
+#ifdef MONO_ARCH_HAVE_SWIFTCALL
+		if (mono_method_signature_has_ext_callconv (sig, MONO_EXT_CALLCONV_SWIFTCALL)) {
+			MonoClass *swift_self = mono_class_try_get_swift_self_class ();
+			MonoClass *swift_error = mono_class_try_get_swift_error_class ();
+			MonoClass *swift_error_ptr = mono_class_create_ptr (m_class_get_this_arg (swift_error));
+			MonoClass *klass = mono_class_from_mono_type_internal (sig->params [i]);
+			if (klass == swift_self && sig->pinvoke) {
+				guint32 old_gr = gr;
+				if (gr >= PARAM_REGS)
+					gr--;
+				add_valuetype (sig, ainfo, ptype, FALSE, &gr, &fr, &stack_size);
```

Comment:
This looks way more confusing that on arm64 case. We should probably just initialize ainfo explicitly as arg in ireg.

Reply:
Ok, same as in arm64 case.